### PR TITLE
Convert samples to use UserSecrets.

### PR DESCRIPTION
**Pull request by Tratcher**
#593 @HaoK @PinpointTownes @muratg

### CODE REVIEW

**Comment by PinpointTownes** [comment]
Personally, I'm not fundamentally opposed to using the user secret thingy, but you need to add a README somewhere with all the `client_id/client_secret` or nobody will be able to use the samples without registering an app with the social providers.

**Comment by Tratcher** [comment]
How many of them work for you today? Most of them only work with my login because the associated apps are in dev mode. You need your own configurations anyways.

**Comment by PinpointTownes** [comment]
The last time I tried, Google/Twitter/GitHub all worked like a charm, but Facebook and MSFT (but this one is terribly painful...). I'll give them another try to confirm.

**Comment by PinpointTownes** [comment]
Confirmed, only FB and MSFT don't work. If you want to avoid hardcoding the client credentials, why not using a JSON config file and adding a comment indicating that the user secret store is recommended on non-sample apps?

File: samples/JwtBearerSample/Startup.cs [hunk 1]
```diff
@@ -53,8 +64,8 @@ public void Configure(IApplicationBuilder app)
                 options.AutomaticAuthenticate = true;
                 options.AutomaticChallenge = true;
                 // You also need to update /wwwroot/app/scripts/app.js
-                options.Authority = "https://login.windows.net/tratcheroutlook.onmicrosoft.com";
-                options.Audience = "63a87a83-64b9-4ac1-b2c5-092126f8474f";
+                options.Authority = Configuration["jwt:authority"];
+                options.Audience = Configuration["jwt:audiance"];
```
Comment (PinpointTownes) [inline samples/JwtBearerSample/Startup.cs]: `jwt:audiance` -> `jwt:audience`

File: samples/JwtBearerSample/Startup.cs [hunk 2]
```diff
@@ -12,6 +13,16 @@ namespace JwtBearerSample
...
     {
+        public Startup()
+        {
+            Configuration = new ConfigurationBuilder()
+                .AddEnvironmentVariables()
+                .AddUserSecrets()
+                .Build();
+        }
+
+        public IConfiguration Configuration { get; set; }
```
Comment (PinpointTownes) [inline samples/JwtBearerSample/Startup.cs]:
```csharp
public IConfiguration Configuration { get; } = new ConfigurationBuilder()
    .AddEnvironmentVariables()
    .AddUserSecrets()
    .Build();
```

:trollface:

**Comment by Eilon** [comment]
@blowdart this good?

**Comment by blowdart** [comment]
Sure

File: samples/JwtBearerSample/Startup.cs (hunk 2 above)
Comment (Tratcher) [inline samples/JwtBearerSample/Startup.cs]: I'm going to leave it because it's closer to the template.

**Comment by Tratcher** [comment]
Updated.

**Pull request merged**